org.eclipse.buildship.ui/src/main/java/org/eclipse/buildship/ui/view/execution/UpdateExecutionPageJob.java

```diff
@@ -34,15 +33,18 @@
 
     private static final int REPEAT_DELAY = 100;
 
-    private final Map<OperationDescriptor, OperationItem> operationItems;
+    private final Set<OperationItem> operationItems;
+    private final Set<OperationItem> removeableItems;
+
     private volatile boolean running;
     private final TreeViewer treeViewer;
 
     public UpdateExecutionPageJob(TreeViewer treeViewer) {
         super("Updating duration of non-finished operations");
 
         this.treeViewer = treeViewer;
-        this.operationItems = Maps.newHashMap();
+        this.operationItems = Sets.newConcurrentHashSet();
```

Comment:
Thanks for the explanation. Unfortunately, the synchronization does cause measurable overhead. I've rewritten the class such that only one `ConcurrentMap` is used and the possible race conditions are eliminated.